### PR TITLE
Fix an unreliable test.

### PR DESCRIPTION
**Pull request by marcin-ro**

### CODE REVIEW

File: integration/full_stack_test.go [hunk 1]
```diff
@@ -166,6 +164,19 @@ func verifyBuildFilesExist(t *testing.T, buildPath string) {
 	}
 }
 
+func waitUntilAvailable(addr string) {
+	for {
+		conn, err := net.Dial("tcp", addr)
+		if conn != nil {
+			// Silently ignore the err
+			_ = conn.Close()
```
Comment (Holek) [inline integration/full_stack_test.go:172]: maybe this here, so we don't bombard the test server with connection attempts?
```suggestion
			_ = conn.Close()
			time.Sleep(time.Second)
```

Comment (Holek) [reply integration/full_stack_test.go:172]: Also, maybe we could add a timer that fails after x tries (like 10 tries?)

File: integration/full_stack_test.go [hunk 2]
```diff
@@ -166,6 +164,19 @@ func verifyBuildFilesExist(t *testing.T, buildPath string) {
 	}
 }
 
+func waitUntilAvailable(addr string) {
+	for {
+		conn, err := net.Dial("tcp", addr)
+		if conn != nil {
```
Comment (Holek) [inline integration/full_stack_test.go:170]: shouldn't that be `err != nil`?

File: integration/full_stack_test.go (hunk 1 above)
Comment (marcin-ro) [reply integration/full_stack_test.go:172]: D'oh, of course.

Comment (marcin-ro) [reply integration/full_stack_test.go:172]: > Also, maybe we could add a timer that fails after x tries (like 10 tries?)
I don't want to worry about that.  Tests have a top-level time limit.

File: integration/full_stack_test.go (hunk 2 above)
Comment (marcin-ro) [reply integration/full_stack_test.go:170]: Simplified.

**Pull request merged**